File: python-algorithms/src/monotonic_array.py
from typing import List

class MonotonicArray:
    def isMonotonic(self, nums: List[int]) -> bool:
        isMonotonic = True
        isAscending = None
        for i in range(0, len(nums) - 1):
            if(nums[i] > nums[i+1]):
                if(isAscending == True):
                    return False
                isAscending = False               
            if(nums[i] < nums[i+1]):
                if(isAscending == False):
                    return False
                isAscending = True

        return isMonotonic

File: java-algorithms/src/main/java/com/omsprog/algorithms/lc/MonotonicArray.java
package com.omsprog.algorithms.lc;

public class MonotonicArray {
    public boolean isMonotonic(int[] nums) {
        int p = nums[0];
        int f = 0; // 0 -> not set, 1 -> increasing, 2 -> decreasing

        for(int i = 1; i < nums.length; i++) {
            int n = nums[i];

            if(f == 0) {
                if(n > p) f = 1;
                else if (n < p) f = 2;
            } else {
                if(n > p && f == 2) return false;
                else if(n < p && f == 1) return false;
            }
            p = nums[i];
        }

        return true;
    }
}
